x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -126,17 +133,30 @@ export async function getPackageInfo({
   pkgVersion,
   skipArchive = false,
   ignoreUnverified = false,
+  prerelease,
 }: {
   savedObjectsClient: SavedObjectsClientContract;
   pkgName: string;
   pkgVersion: string;
   /** Avoid loading the registry archive into the cache (only use for performance reasons). Defaults to `false` */
   skipArchive?: boolean;
   ignoreUnverified?: boolean;
+  prerelease?: boolean;
 }): Promise<PackageInfo> {
+  // if prerelease param not defined, query from settings
+  if (prerelease === undefined) {
```

Comment:
Currently added an optional `prerelease` parameter to `getPackageInfo` API, so that the UI can query the latest GA and latest Prerelease versions: https://github.com/elastic/kibana/blob/0c8087287a3c80a07c8d90026bd9615715beb6a7/x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/index.tsx#L185-L201
If not provided, the logic reads the `settings` to determine the prerelease flag value.

I am wondering if it would be better to create a new API instead that returns all versions of a package instead of using `getPackageInfo` API for that.